src/functions/asyncCredential/asyncCredentialHandler.ts

```diff
@@ -111,7 +112,37 @@ export async function lambdaHandler(
   }
 
   const clientCredentials =
-    clientCredentialResponse.value as IClientCredentials;
+    clientCredentialResponse.value as IRegisteredClientCredentials;
+
+  const validateClientCredentialsResponse = clientCredentialsService.validate(
+    clientCredentials,
+    JSON.parse(requestBody),
+  );
+  if (validateClientCredentialsResponse.value != null) {
+    return badRequestResponse({
+      error: "invalid_request",
+      errorDescription: validateClientCredentialsResponse.value,
+    });
+  }
+
+  let parsedRequestBody: IClientCredentials;
+  try {
+    parsedRequestBody = JSON.parse(requestBody);
```

Comment:
The check to parse JSON should be much further up in this lambda. Please may we move it to the request body validation function if it's not there already,

Reply:
Updated.